utilities/shared_disk.py

```
@@ -253,58 +291,535 @@ def verify_shared_disk_data(
+        owner_name: Owner VM name (for error messages).
+        vm_names: All VM names in the plan.
+
+    Returns:
+        Disk serial string (backing UUID without dashes, lowercase).
+
+    Raises:
+        ValueError: If no shared VMDK found or backing UUID unavailable.
+    """
+    shared_vmdks = source_provider.find_shared_vmdk_paths(vm_names)
+    if not shared_vmdks:
+        raise ValueError(f"No shared VMDKs found between VMs: {vm_names}")
```

Ruff (0.15.18)

[warning] Avoid specifying long messages outside the exception class

(TRY003)